arangod/RestServer/TtlFeature.cpp

```diff
@@ -0,0 +1,610 @@
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, "expecting object for statistics");
+  }
+
+  if (other.hasKey("runs")) {
+    runs += other.get("runs").getNumericValue<uint64_t>();
+  }
+  if (other.hasKey("documentsRemoved")) {
+    documentsRemoved += other.get("documentsRemoved").getNumericValue<uint64_t>();
+  }
+  if (other.hasKey("limitReached")) {
+    limitReached += other.get("limitReached").getNumericValue<uint64_t>();
+  }
+  return *this;
+}
+  
+void TtlStatistics::toVelocyPack(VPackBuilder& builder) const {
+  builder.openObject();
+  builder.add("runs", VPackValue(runs));
+  builder.add("documentsRemoved", VPackValue(documentsRemoved));
+  builder.add("limitReached", VPackValue(limitReached));
+  builder.close();
+}
+
+void TtlProperties::toVelocyPack(VPackBuilder& builder, bool isActive) const {
+  builder.openObject();
+  builder.add("frequency", VPackValue(frequency)); 
+  builder.add("maxTotalRemoves", VPackValue(maxTotalRemoves));
+  builder.add("maxCollectionRemoves", VPackValue(maxCollectionRemoves));
+  builder.add("onlyLoadedCollections", VPackValue(onlyLoadedCollections));
+  builder.add("active", VPackValue(isActive));
+  builder.close();
+}
+
+Result TtlProperties::fromVelocyPack(VPackSlice const& slice) {
+  if (!slice.isObject()) {
+    return Result(TRI_ERROR_BAD_PARAMETER, "expecting object for properties");
+  }
+
+  try {
+    uint64_t frequency = this->frequency;
+    uint64_t maxTotalRemoves = this->maxTotalRemoves;
+    uint64_t maxCollectionRemoves = this->maxCollectionRemoves;
+    uint64_t onlyLoadedCollections = this->onlyLoadedCollections;
+
+    if (slice.hasKey("frequency")) {
+      if (!slice.get("frequency").isNumber()) {
+        return Result(TRI_ERROR_BAD_PARAMETER, "expecting numeric value for frequency");
+      }
+      frequency = slice.get("frequency").getNumericValue<uint64_t>();
+    }
+    if (slice.hasKey("maxTotalRemoves")) {
+      if (!slice.get("maxTotalRemoves").isNumber()) {
+        return Result(TRI_ERROR_BAD_PARAMETER, "expecting numeric value for maxTotalRemoves");
+      }
+      maxTotalRemoves = slice.get("maxTotalRemoves").getNumericValue<uint64_t>();
+    }
+    if (slice.hasKey("maxCollectionRemoves")) {
+      if (!slice.get("maxCollectionRemoves").isNumber()) {
+        return Result(TRI_ERROR_BAD_PARAMETER, "expecting numeric value for maxCollectionRemoves");
+      }
+      maxCollectionRemoves = slice.get("maxCollectionRemoves").getNumericValue<uint64_t>();
+    }
+    if (slice.hasKey("onlyLoadedCollections")) {
+      if (!slice.get("onlyLoadedCollections").isBool()) {
+        return Result(TRI_ERROR_BAD_PARAMETER, "expecting boolean value for onlyLoadedCollections");
+      }
+      onlyLoadedCollections = slice.get("onlyLoadedCollections").getBool();
+    }
+
+    this->frequency = frequency;
+    this->maxTotalRemoves = maxTotalRemoves;
+    this->maxCollectionRemoves = maxCollectionRemoves;
+    this->onlyLoadedCollections = onlyLoadedCollections;
+
```

Comment:
```suggestion
frequency = VelocyPackHelper::getNumericValue<uint64_t>(slice, "frequency", frequency);
maxTotalRemoves = VelocyPackHelper::getNumericValue<uint64_t>(slice, "maxTotalRemoves", maxTotalRemoves);
maxCollectionRemoves = VelocyPackHelper::getNumericValue<uint64_t>(slice, "maxCollectionRemoves", maxCollectionRemoves);
onlyLoadedCollections = VelocyPackHelper::getBooleanValue(slice, "onlyLoadedCollections", onlyLoadedCollections);
```

Reply:
Instead of the whole mess inside the `try` block.

Reply:
Mmmh... This is not quite the same.
My approach validates the input parameters and returns errors if they are wrong.
Yours doesn't, and silently uses the default values instead.